### ISSUE TITLE
Styled components styles don't show up on production build

### ISSUE THREAD

**Issue by AmodeusR** [styled-components/styled-components#3699]
## Environment

The instructions didn't work.

## Reproduction
I just built the project. Here's the repo if you want to see it: [Repo link](https://github.com/AmodeusR/FM--time-tracking-dashboard)

## Expected Behavior
Styles show up on prod build.

## Actual Behavior
Styles don't show up on prod build, only in dev enviroment.

## Others
I'm just using Styled Components to style the card's background because I needed it to be dynamically defined. The classes are generated and applied, but without any styling. I already tried Styled Components as a dependency, as a dev dependency, tried ENV file in the root and src folder with the "Disable speedy" thing I heard around and some other stuff for **hours** but nothing.

Could anyone help me with this issue?

**Comment by AmodeusR** [comment]
Just confirmed it had to do with the build tool I was using and some tricky things I was doing and not with Styled Components 😅

**Comment by bayerlse** [comment]
@AmodeusR which build tool did you use, and how did you fix it? I'm currently running into this issue as well :/

**Comment by AmodeusR** [comment]
It has passed so many time that I don't remember very well, sorry for not explaining what I did to solve it 😥

But I believe it having to do with the fact that when it was built, the build tool added some strings to the icon I was trying to take its name from to define dynamically the background color of a card, thus the reference to the class not working. That happened because it was based in an imported asset, and build tools generally add some random strings to it.

What I did was stop trying taking the name from the icon itself which would have the random string and created a separate variable to get the name of the icon independently, which wouldn't have the random string attached to it.

You [can see it here](https://github.com/AmodeusR/FM--time-tracking-dashboard/commit/14e6486b8b7f169e90aa63fa906b25961ad59e4b#diff-d39fa35ffc5ea93f748103ec4fe502dcdea103b3b7923020b7d4273a9f4c58f0). _iconRef_ before would try to filter out the name from the icon asset, but because of the extra characters wouldn't work in the built project. I made the _iconRef_ then as a normalizer and used _cardBackground_ to set the styles I wanted.
